palantirnet/federated-search-react
# Opensource 34 namespace CSS

Completes [OPENSOURCE-34](https://palantir.atlassian.net/browse/OPENSOURCE-34)

## Description:

Ensures all CSS is namespaced within the Federated Search Application. The ID of the main app element has been renamed `#fs-root` so that it is a more unique identifier than previous `#root` and all previous base element styles are now applied by a `globals` mixin to `#fs-root`. All component, layout and helper classes now have the prefix `fs-` added to them for more specific namespacing. 

## To Test 

Review visually to make sure the app still looks the same. 
Review the exported styles.css and scss files file and verify:
- There are no elements globally styled without a class. All global styles should now be scoped within the main `#fs-root` class using the `globals` mixin. 
- All classes start with `fs-` prefix. There are a number of exceptions I will note below:

  - the date range styles in src/components/range-facet/_react-dates.scss and src/components/range-facet/_react-dates-overrides.scss appeared to be for styling a specific react component, so I left them.
  - the classes beginning with `.slide-` in src/components/text-search/_fs-autocomplete.scss because I could not find them in use. I think they can be removed and I left a comment suggesting it. 
  - I worked through all the scss files and updated classes throughout the rest of the code based on changes there. There are still a number of classes that do not have the `fs-` prefix in the markup but they are not being used for styles. These should probably be reviewed and renamed for consistency or removed if not needed.
